### PR TITLE
add api call getSubnet

### PR DESCRIPTION
**Pull request by najeal**
## Why this should be merged

It is related to issue #1255 

## How this works

It gets subnet tx from an ID.

## How this was tested

Tested in service_test.go through mock.

### CODE REVIEW

**Comment by StephenButtolph** [comment]
Thanks for opening this!

I'm wondering if it would be sufficient for the added API to only be `GetSubnetTransformation`, since `GetTx` could already be used to return the `CreateSubnetTx`.

Although it might be a pain for users to be able to tell the difference between a permissioned subnet and a networking error in that case.

**Comment by najeal** [comment]
The idea was to have a generic call to get `subnet` then depending of the transaction being either `CreateSubnetTx` or `TransformSubnetTx` the user knows if the subnet is an elastic one or not.

Yes the user can already use `GetTx()` to get the `CreateSubnetTx`, but at this moment the user still needs to do an extra call to `getSubnet(or GetSubnetTransformationTx)` to know if it has been transformed.

What do you think @StephenButtolph It's up to you.

File: vms/platformvm/service.go [hunk 1]
```diff
@@ -62,22 +62,23 @@ const (
...
+	errStartTimeTooLate          = errors.New("start time is too far in the future")
+	errNamedSubnetCantBePrimary  = errors.New("subnet validator attempts to validate primary network")
+	errNoAmount                  = errors.New("argument 'amount' must be > 0")
+	errMissingName               = errors.New("argument 'name' not given")
+	errMissingVMID               = errors.New("argument 'vmID' not given")
+	errMissingBlockchainID       = errors.New("argument 'blockchainID' not given")
+	errMissingPrivateKey         = errors.New("argument 'privateKey' not given")
+	errStartAfterEndTime         = errors.New("start time must be before end time")
+	errStartTimeInThePast        = errors.New("start time in the past")
+	errUnexpectedTransactionType = fmt.Errorf("unexpected transaction type")
```
Comment (patrick-ogrady) [inline vms/platformvm/service.go:81]: nit: use `errors.New` here

Comment (najeal) [reply vms/platformvm/service.go]: done

File: vms/platformvm/service.go [hunk 2]
```diff
@@ -527,6 +528,47 @@ type GetSubnetsResponse struct {
...
+func (s *Service) GetSubnet(_ *http.Request, args *api.GetSubnetArgs, response *api.GetTxReply) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getSubnet"),
+	)
+
+	subnetID := args.SubnetID
+
+	var tx *txs.Tx
+	var err error
```
Comment (ceyonur) [inline vms/platformvm/service.go:540]: can we remove this and assign to them below like `tx, err := s.vm.state.GetSubnetTransformation(subnetID)`

File: vms/platformvm/service.go [hunk 3]
```diff
@@ -527,6 +528,47 @@ type GetSubnetsResponse struct {
...
+	tx, err = s.vm.state.GetSubnetTransformation(subnetID)
+	if err != nil && err != database.ErrNotFound {
+		return fmt.Errorf("couldn't get transformed subnet: %w", err)
+	}
+	if err != nil && err == database.ErrNotFound {
```
Comment (ceyonur) [inline vms/platformvm/service.go:545]: We don't need to recheck `err !=nil` here before `err == database.ErrNotFound`

File: vms/platformvm/service.go [hunk 4]
```diff
@@ -527,6 +528,47 @@ type GetSubnetsResponse struct {
...
+		// subnet may not be transformed
+		tx, _, err = s.vm.state.GetTx(subnetID)
+		if err != nil {
+			return fmt.Errorf("couldn't get tx: %w", err)
+		}
+		_, ok := tx.Unsigned.(*txs.CreateSubnetTx)
+		if !ok {
+			return fmt.Errorf("%w expected *txs.CreateSubnetTx but got %T", errUnexpectedTransactionType, tx.Unsigned)
+		}
```
Comment (ceyonur) [inline vms/platformvm/service.go:554]: could be simplified like:
```
if _, ok := tx.Unsigned.(*txs.CreateSubnetTx); !ok {
   return fmt.Errorf("%w expected *txs.CreateSubnetTx but got %T", errUnexpectedTransactionType, tx.Unsigned)
}
```

**Review by ceyonur — COMMENTED** [review]
just some nits

File: vms/platformvm/service.go (hunk 4 above)
Comment (najeal) [reply vms/platformvm/service.go]: done

File: vms/platformvm/service.go (hunk 3 above)
Comment (najeal) [reply vms/platformvm/service.go]: done

File: vms/platformvm/service.go (hunk 2 above)
Comment (najeal) [reply vms/platformvm/service.go]: done

**Comment by najeal** [comment]
Any update?

**Comment by najeal** [comment]
Closing this PR.  #2704 is merged and resolves the issue

**Pull request closed without merging**